permit_jsligo/src/main.jsligo

```diff
@@ -0,0 +1,192 @@
+      let {owner,token_id,amount_} = mint_or_burn_param;
+      FA2.Ledger.decrease_token_amount_for_user(ledger, owner, token_id, amount_);
+   };
+   
+   let ledger = List.fold_left(process_one, s.ledger, lst);
+   let s = FA2.Storage.set_ledger(s, ledger);
+   [(list([]) as list<operation>), s]
+};
+
+//(* TZIP-17 *)
+let permit = ([lst, s] : [list<permit_params>, storage]) : storage => {
+    let process_permit = ([ext, permit] : [Extension.t, permit_params]) : Extension.t => {
+        let [pub_key, [sig, hash_]] = permit;
+        let packed = Bytes.pack([(Tezos.get_chain_id()), Tezos.get_self_address()], [ext.counter, hash_]);
+        if (Crypto.check(pub_key, sig, packed)) {
+            let sender_ = Tezos.address (Tezos.implicit_account (Crypto.hash_key(pub_key)));
+            let permit_key = [sender_, hash_];
+            match (Big_map.find_opt(permit_key, ext.permits), {
+                None: () => Extension.add_permit(ext, permit_key),
+                Some: (submission_timestamp: timestamp) => {
+                    let _ = Extension._check_not_expired(s.extension, submission_timestamp, permit_key);
+                    Extension.update_permit(ext, permit_key)
+                }
+            })
+        } else { 
+        //([%Michelson ({| { FAILWITH } |} : string * bytes -> Extension.t)]) (Errors.missigned, packed)
+            (Michelson `{FAILWITH}` as ((n: [string, bytes]) => Extension.t)) ([Errors.missigned, packed])
+        }
+    };
+    let extension = List.fold_left(process_permit, s.extension, lst);
+    return { ...s, extension : extension }
+};
+
+//(* TZIP-17 *)
+let set_expiry = ([p, s] : [expiry_params, storage]) : storage => {
+    let [user_address, [seconds, permit_hash_opt]] = p;
+    if (seconds > s.extension.max_expiry) {
+        return (failwith(Errors.max_seconds_exceeded) as storage)
+    } else { 
+        if (Tezos.get_sender() != user_address) {
+            return (failwith(Errors.forbidden_expiry_update) as storage)
+        } else {
+            let new_storage =
+                match (permit_hash_opt, {
+                    None: () => {
+                        return {...s, extension : { ...s.extension, 
+                            user_expiries : Big_map.add(user_address, (Some(seconds)), s.extension.user_expiries) 
+                        }};                        
+                    },
+                    Some: (permit_hash: bytes) => {
+                        return {
+                            ...s, extension : {...s.extension, permit_expiries : Big_map.add(
+                                [user_address, permit_hash],
+                                (Some(seconds)),
+                                s.extension.permit_expiries)
+                        }}
+                    }
+                });
+            new_storage
+        }
+    }
+};
+
+//(* TZIP-17 implementation of TZIP-12 Transfer *)
+let transfer_permitted = ([t, s]: [FA2.transfer, storage]) : storage => {
```

Comment:
plz take a look at Melwynn feedback on my MR https://github.com/ligolang/permit-cameligo/pull/1#discussion_r937266417
this is based on old code